Makefile

```diff
@@ -37,7 +37,11 @@ CMAKE_ARGS ?=
 VERBOSE ?=
 FORCE_CLANG_FORMAT ?=
 
-pyextsuffix := $(shell python3-config --extension-suffix)
+WHICH_PYTHON := $(shell which python)
+REALPATH_PYTHON := $(shell realpath $(WHICH_PYTHON))
+DIRNAME_PYTHON := $(shell dirname $(REALPATH_PYTHON))
+
+pyextsuffix := $(shell $(DIRNAME_PYTHON)/python3-config --extension-suffix)
```

Comment:
FWIW, this will bring the build flow the support of using python virtual environment.

For example, in my case it is:
- `which python`: /home/tai271828/.cache/pypoetry/virtualenvs/modmesh-venv-py3-9-13-LgJ7FNfF-py3.9/bin/python
- `realpath`: /home/tai271828/.pyenv/versions/3.9.13/bin/python3.9
- `dirname`: /home/tai271828/.pyenv/versions/3.9.13/bin/

and additionally:
```
$ ls -alh /home/tai271828/.cache/pypoetry/virtualenvs/modmesh-venv-py3-9-13-LgJ7FNfF-py3.9/bin/python
lrwxrwxrwx 1 tai271828 tai271828 49 Dec  9 21:46 /home/tai271828/.cache/pypoetry/virtualenvs/modmesh-venv-py3-9-13-LgJ7FNfF-py3.9/bin/python -> /home/tai271828/.pyenv/versions/3.9.13/bin/python
```

Reply:
Please summary in a comment line that the treatment is for virtual env.  It should also be mentioned that a virtual env is strongly discouraged in developing modmesh.  We do not actively resolve bugs related to any virtual env including venv or conda.

Reply:
Will do!